python/lsst/pipe/tasks/computeExposureSummaryStats.py

```diff
@@ -830,3 +891,64 @@ def compute_ap_corr_sigma_scaled_delta(
         ap_corr_sigma_scaled_delta = (np.max(ap_corr_good) - np.min(ap_corr_good))/psfSigma
 
     return ap_corr_sigma_scaled_delta
+
+
+def compute_magnitude_limit(
+        psfArea,
+        skyBg,
+        zeroPoint,
+        readNoise=0.0,
+        gain=1.0,
+        snr=5
+):
+    """Compute the expected point-source magnitude limit at a given
+    signal-to-noise ratio given the exposure-level metadata. Based on
+    the signal-to-noise formula provided in SMTN-002 (see LSE-40 for
+    more details on the calculation).
+
+      SNR = C / sqrt( C/g + (B/g + sigma_inst**2) * neff )
+
+    where C is the counts from the source, B is counts from the (sky)
+    background, sigma_inst is the instrumental (read) noise, neff is
+    the effective size of the PSF, and g is the gain in e-/ADU.
+    Note that input values of skyBg, zeroPoint, and readNoise should
+    all consistently be in electrons or ADU.
+
+    Parameters
+    ----------
+    psfArea  : `float`
+        The effective area of the PSF [pix].
+    skyBg     : `float`
+        The sky background counts for the exposure [ADU or e-].
+    zeroPoint : `float`
+        The zeropoint (includes exposure time) [ADU or e-].
+    readNoise : `float`, optional
+        The instrumental read noise for the exposure [ADU or e-].
+    gain      : `float`, optional
+        The instrumental gain for the exposure [e-/ADU]. The gain should
+         be 1.0 if the skyBg, zeroPoint, and readNoise are in e-.
+    snr       : `float`, optional
+        Signal-to-noise ratio at which magnitude limit is calculated.
+
+    Returns
+    -------
+    magnitude_limit : `float`
+        The expected magnitude limit at the given signal to noise.
+    """
+    # Effective number of pixels within the PSF calculated directly
+    # from the PSF model.
+    neff = psfArea
+
+    # Instrumental noise (read noise only)
+    sigma_inst = readNoise
```

Comment:
This file is nominally in camelCase, so please don't start using snake_case randomly inside. Here and in a few other places, _e.g._ `magnitude_limit` below, etc.

Reply:
Ok, I was following the convention (to the extent that it exists) of the other "compute" functions, which seem to internally use snake case. It seems that camelCase is really being used for the properties that propagate outside this file (i.e., psfSigma, skyBg, etc.)

Reply:
Oh, what a mess. OK, whatever you like this. Not my area of the codebase anyway tbh.